src/solvers/dgsem_tree/dg_1d.jl

```diff
@@ -332,11 +412,45 @@ function calc_interface_flux!(surface_flux_values,
   end
 end
 
-# TODO: MHD in 1D
-# function calc_interface_flux!(surface_flux_values, mesh::TreeMesh{1},
-#                               nonconservative_terms::Val{true}, equations,
-#                               dg::DG, cache)
-# end
+function calc_interface_flux!(surface_flux_values,
+                              mesh::TreeMesh{1},
+                              nonconservative_terms::Val{true}, equations,
+                              surface_integral, dg::DG, cache)
+  @unpack surface_flux = surface_integral
+  @unpack u, neighbor_ids, orientations = cache.interfaces
+
+  @threaded for interface in eachinterface(dg, cache)
+    # Get neighboring elements
+    left_neighbor  = neighbor_ids[1, interface]
+    right_neighbor = neighbor_ids[2, interface]
+
+    # Determine interface direction with respect to elements:
+    # orientation = 1: left -> 2, right -> 1
+    left_direction  = 2 * orientations[interface]
+    right_direction = 2 * orientations[interface] - 1
+
+    # Call pointwise Riemann solver
+    u_ll, u_rr = get_surface_node_vars(u, equations, dg, interface)
+    f = surface_flux(u_ll, u_rr, orientations[interface], equations)
+
+    # Compute the nonconservative numerical "flux" along an interface
+    # Done twice because left/right orientation matters så
+    # 1 -> primary element and 2 -> secondary element
+    # See Bohm et al. 2018 for details on the nonconservative diamond "flux"
+
+    # Call pointwise nonconservative term
+    noncons_primary   = noncons_interface_flux(u_ll, u_rr, orientations[interface], :weak, equations)
```

Comment:
This routine does not exist for the MHD, so currently this version of the `calc_interface_flux` is only called via your new tutorial, right?

Reply:
Exactly :+1: